components/learn/modules/ROOT/pages/unit-testing.adoc

```diff
@@ -0,0 +1,174 @@
+const MyContract = contract.fromArtifact('MyContract');
+```
+
+For detailed information on these exported values and their usage, refer to Test Environment's https://github.com/OpenZeppelin/openzeppelin-test-environment/blob/master/docs/modules/ROOT/pages/api.adoc[API reference].
+
+[[writing-unit-tests]]
+== Writing Unit Tests
+
+In order to actually run your tests, you will need to also install a JavaScript _test runner_. You are free to use any of the https://github.com/OpenZeppelin/openzeppelin-test-environment/blob/master/docs/modules/ROOT/pages/setup.adoc#test-runners[recommended ones]: for this guide, we'll pick https://mochajs.org/[Mocha] with https://www.chaijs.com/[Chai] assertions.
+
+```bash
+$ npm install --save-dev mocha chai
+```
+
+Create a `test` directory: this is where you will keep your test files. These are best structured by mirroring xref:write-contracts#setting-up-a-solidity-project[the `contracts` directory]: for each `.sol` file there, create a corresponding `.test.js` file.
+
+Time to write our first tests! These will test properties of the `SetAndGet` contract xref:write-contracts#set-and-get-contract[from previous guides].
+
+```javascript
+// Load dependencies
+const { accounts, contract } = require('@openzeppelin/test-environment');
+const { expect } = require('chai');
+
+// Load compiled artifacts
+const SetAndGet = contract.fromArtifact('SetAndGet');
+
+// Start test block
+describe('SetAndGet', function () {
+  const [ owner ] = accounts;
+
+  beforeEach(async function () {
+    // Deploy a new contract for each test
+    this.contract = await SetAndGet.new({ from: owner });
+  });
+
+  // Test case
+  it('get returns a value previously set', async function () {
+    // Set a value
+    await this.contract.set(42, { from: owner });
+
+    // Test the returned value is the same one - values from the blockchain are
+    // often read as strings
+    expect((await this.contract.get()).toString()).to.equal('42');
+  });
+});
+```
+
+TIP: Many books have been written about how to structure unit tests: for a quick reference, check out the https://github.com/OpenZeppelin/openzeppelin-contracts/tree/master/test[tests for OpenZeppelin Contracts], or the following https://medium.com/coinmonks/how-to-test-ethereum-smart-contracts-ac28fa852281[guide for ERC20 tests].
+
+We are now ready to run our tests! The best way to do this is by adding a `test` script to your `package.json`:
+
+[source,diff]
+----
+ 'scripts': {
++  'test': 'mocha --exit --recursive test'
+ }
+----
+
+With this, running `npm test` will execute all tests in the `test` directory, checking that your contracts work the way you meant them too.
+
+TIP: Don't forget to re-compile your contracts if you make changes to them!
+
+It's also a very good idea at this point to set up a Continuous Integration service such as https://circleci.com/[CircleCI] to make your tests run automatically every time you commit your code to GitHub.
+
```

Comment:
Should we add a paragraph explaining that it's possible to configure the environment set up by test-env via a config file, in case the reader needs more complex set ups, and link to the reference?

Reply:
I tried putting this paragraph in the previous section and it didn't fit, but this _is_ a good spot for it. Thanks!